client/ayon_core/hosts/maya/plugins/publish/validate_frame_range.py

```diff
@@ -53,12 +53,26 @@ def process(self, instance):
             )
             return
 
-        frame_start_handle = int(context.data.get("frameStartHandle"))
-        frame_end_handle = int(context.data.get("frameEndHandle"))
-        handle_start = int(context.data.get("handleStart"))
-        handle_end = int(context.data.get("handleEnd"))
-        frame_start = int(context.data.get("frameStart"))
-        frame_end = int(context.data.get("frameEnd"))
+        # Get frame information from task entity
+        # NOTE: If there is no task override then the asset
+        # value is automatically returned instead
+        task_entity = instance.context.data["taskEntity"]
+        frame_start_handle = task_entity["attrib"]["frameStart"] - \
+            task_entity["attrib"]["handleStart"]
+        frame_end_handle = task_entity["attrib"]["frameEnd"] + \
+            task_entity["attrib"]["handleEnd"]
+        handle_start = task_entity["attrib"]["handleStart"]
+        handle_end = task_entity["attrib"]["handleEnd"]
+        frame_start = task_entity["attrib"]["frameStart"]
+        frame_end = task_entity["attrib"]["frameEnd"]
+
+        # Get frame information from asset context
+        # frame_start_handle = int(context.data.get("frameStartHandle"))
+        # frame_end_handle = int(context.data.get("frameEndHandle"))
+        # handle_start = int(context.data.get("handleStart"))
+        # handle_end = int(context.data.get("handleEnd"))
+        # frame_start = int(context.data.get("frameStart"))
+        # frame_end = int(context.data.get("frameEnd"))
```

Comment:
It might be nice to query this data in a separate class method so the logic isn't duplicated between here and `repair`

E.g.

```python
        def get_instance_task_frame_range(instance):
            # Get frame information from task entity
            # NOTE: If there is no task override then the asset
            # value is automatically returned instead
            attrib = instance.context.data["taskEntity"]["attrib"]
            handle_start = attrib["handleStart"]
            handle_end = attrib["handleEnd"]
            frame_start = attrib["frameStart"]
            frame_end = attrib["frameEnd"]
            frame_start_handle = frame_start - handle_start
            frame_end_handle = frame_end + handle_end
```

We could potentially even improve the readability by having a FrameRange dataclass.
For example:
```python
@dataclasses.dataclass
class FrameRange:
    """Frame range with handles.

    The frame range excludes the handles - and thus the handles are outside
    of the frame range. To get the inclusive start and end frame use
    `frame_start_handle` and `frame_end_handle`.

    """
    frame_start: int
    frame_end: int
    handle_start: int
    handle_end: int

    @property
    def frame_start_handle(self) -> int:
        return self.frame_start - self.handle_start

    @property
    def frame_end_handle(self) -> int:
        return self.frame_end + self.handle_end


# Example usage
frames = FrameRange(frame_start=1001,
                    frame_end=1010,
                    handle_start=5,
                    handle_end=10)

print(frames.frame_start_handle)
```

Then the method becomes:
```python
        def get_instance_task_frame_range(instance):
            # Get frame information from task entity
            # NOTE: If there is no task override then the asset
            # value is automatically returned instead
            attrib = instance.context.data["taskEntity"]["attrib"]
            return FrameRange(
                frame_start=attrib["frameStart"],
                frame_end=attrib["frameEnd"],
                handle_start=attrib["handleStart"],
                handle_end=attrib["handleEnd"],
            )
```

Reply:
Furthermore I still think this should check the frame range against the entity of the instance - not the current asset so that you still validate to the asset you're publishing to, not where you're publishing from.

So likely we'll also need to collect the task entity for the instance - and use that one here instead.

Reply:
I implemented the get_instance_task_frame_range, and FrameRange object as suggested in the validate_frame_range.py.

I'm not sure what you mean by this below. The validation and repair action seem to be doing the right thing in my testing so far.

> Furthermore I still think this should check the frame range against the entity of the instance - not the current asset so that you still validate to the asset you're publishing to, not where you're publishing from.
> 
> So likely we'll also need to collect the task entity for the instance - and use that one here instead.

Reply:
Create two instances in your scene, change folder + task on one of the instances via the publisher UI.
Your one scene, now publishes into two different folder + task entities.

The frame range validation should now validate the frame range of the selected task entity - not from your current context. So make sure in the database that each folder (and its task?) entity has a different frame range set so you can confirm the validation adheres to that actual selected task entity's frame range.

The current logic relies on `instance.context` of which during publishing there's only one - and thus I'm quite confident this currently does not work the way I describe.

Reply:
@BigRoy is it just a matter of storing taskEntity in the. so it gets stored per instance
instance.data["taskEntity"]

Reply:
I updated the collection of taskEntity to happen within collect_instances.py rather than collect_context_entities.py,
https://github.com/ynput/ayon-core/pull/39/files#diff-13c18dd7cb684126254981e14bcf1590c1c5be889bc327ce05bbddc404d2133fR96

Is this the right approach?

Reply:
@BigRoy do you have any further information about approach above?

Reply:
> is it just a matter of storing taskEntity in the instance data. so it gets stored per instance, rather than per context.
> instance.data["taskEntity"]

Yes, along those lines.

Basically the same behavior as is done for "assets" [here](https://github.com/pypeclub/OpenPype/blob/5e56cc4dc31226fc96100e7f5f86ef5699d99bb9/openpype/plugins/publish/collect_anatomy_instance_data.py#L40) basically inheriting the asset entity from context if it matches the context otherwise querying the differing asset entity instead.

Reply:
sorry I do not follow. 

is it possible for you to propose a code change for this?
It's really difficult to follow in written form.